crates/ekore/src/constants.rs

```diff
@@ -71,24 +71,31 @@ pub fn uplike_flavors(nf: u8) -> u8 {
 }
 
 pub struct ChargeCombinations {
-    pub e2avg: f64,
-    pub vue2m: f64,
-    pub vde2m: f64,
-    pub e2delta: f64,
+    pub nf: u8,
 }
 
-pub fn charge_combinations(nf: u8) -> ChargeCombinations {
-    let nu = uplike_flavors(nf) as f64;
-    let nd = (nf as f64) - nu;
-    let e2avg = (nu * eu2 + nd * ed2) / (nf as f64);
-    let vue2m = nu / (nf as f64) * (eu2 - ed2);
-    let vde2m = nd / (nf as f64) * (eu2 - ed2);
-    let e2delta = vde2m - vue2m + e2avg;
-
-    ChargeCombinations {
-        e2avg,
-        vue2m,
-        vde2m,
-        e2delta,
+impl ChargeCombinations {
+    pub fn nu(&self) -> u8 {
+        uplike_flavors(self.nf)
```

Comment:
let's move the function here
```suggestion
        self.nf / 2
```
I think we can drop the exception ...